community/command-line/src/main/java/org/neo4j/commandline/admin/HelpCommand.java

```diff
@@ -38,34 +41,62 @@ public Provider( Usage usage )
         @Override
         public Optional<String> arguments()
         {
-            return Optional.empty();
+            return Optional.of( "[<command>] ");
         }
 
         @Override
         public String description()
         {
-            return "Display this help text.";
+            return "This help text, or help for the command specified in <command>.";
+        }
+
+        @Override
+        public String summary()
+        {
+            return description();
         }
 
         @Override
         public AdminCommand create( Path homeDir, Path configDir, OutsideWorld outsideWorld )
         {
-            return new HelpCommand( usage, outsideWorld::stdOutLine );
+            return new HelpCommand( usage, outsideWorld::stdOutLine, CommandLocator.fromServiceLocator() );
         }
     }
 
     private final Usage usage;
     private final Consumer<String> output;
+    private final CommandLocator locator;
 
-    public HelpCommand( Usage usage, Consumer<String> output )
+    public HelpCommand( Usage usage, Consumer<String> output, CommandLocator locator )
     {
         this.usage = usage;
         this.output = output;
+        this.locator = locator;
     }
 
     @Override
     public void execute( String... args )
     {
-        usage.print( output );
+        if (args.length > 0)
+        {
+            try
+            {
+                AdminCommand.Provider commandProvider = this.locator.findProvider( args[0] );
+                Usage.CommandUsage commandUsage = new Usage.CommandUsage( commandProvider, "neo4j-admin" );
+                commandUsage.print( this.output );
+                this.output.accept( commandProvider.description() );
+                this.output.accept( "" );
+                return;
+            }
+            catch (NoSuchElementException e)
+            {
+                this.output.accept( "Unknown command: " + args[0] );
+                this.output.accept( "" );
```

Comment:
This case should exit non-zero, I think. That is, it needs to throw an exception. How about also arranging for it to print the valid list of commands?